scripts/gui/pages/CpCourseGeneratorFrame.lua

```diff
@@ -113,6 +116,22 @@ function CpCourseGeneratorFrame.createFromExistingGui(gui, guiName)
 	return newGui
 end
 
+function CpCourseGeneratorFrame.registerXmlSchema(xmlSchema, xmlKey)
+	xmlKey = xmlKey .. CpCourseGeneratorFrame.BASE_XML_KEY
+	xmlSchema:register(XMLValueType.INT, xmlKey .. "#mapHotspotValue", "Selected hotspots")
+end
+
+
+function CpCourseGeneratorFrame:loadFromXMLFile(xmlFile, baseKey)
+	self.cpHotspotSettingValue = xmlFile:getValue(
+		string.format("%s%s#mapHotspotValue", baseKey, self.BASE_XML_KEY), 0xFF)
+end
+
+function CpCourseGeneratorFrame:saveToXMLFile(xmlFile, baseKey)
+	xmlFile:setValue(string.format("%s%s#mapHotspotValue", 
+		baseKey, self.BASE_XML_KEY), self.cpHotspotSettingValue)
+end
+
 function CpCourseGeneratorFrame:setInGameMap(ingameMap, hud)
```

Comment:
The hack with the hotspots in this function would deserve a few lines of comment explaining at least the motivation and in general what are we trying to accomplish.